opensrp-giz-malawi/src/main/java/org/smartregister/giz/repository/ReasonForDefaultingRepository.java

```diff
@@ -0,0 +1,111 @@
+        values.put(GizConstants.JsonAssetsHelper.EVENT_DATE, reasonForDefaultingModel.getDateCreated());
+        values.put(GizConstants.JsonAssetsHelper.OTHER_DEFAULTING_REASON, reasonForDefaultingModel.getOtherOutreachDefaultingReason());
+        return values;
+    }
+
+    public ArrayList<ReasonForDefaultingModel> readAllReasonsForDefaulting() {
+        String[] columns = {GizConstants.JsonAssetsHelper.ID, GizConstants.JsonAssetsHelper.OUTREACH_DATE,
+                GizConstants.JsonAssetsHelper.FOLLOWUP_DATE, GizConstants.JsonAssetsHelper.OUTREACH_DEFAULTING_REASON, GizConstants.JsonAssetsHelper.ADDITIONAL_DEFAULTING_NOTES,
+                GizConstants.JsonAssetsHelper.BASE_ENTITY_ID, GizConstants.JsonAssetsHelper.EVENT_DATE, GizConstants.JsonAssetsHelper.OTHER_DEFAULTING_REASON};
+        Cursor cursor = getReadableDatabase().query(TABLE_NAME, columns, null, null, null, null, null, null);
+        ArrayList<ReasonForDefaultingModel> reasonForDefaultingModels = new ArrayList<>();
+
+        try {
+            if (cursor != null && cursor.moveToFirst()) {
+                do {
+                    reasonForDefaultingModels.add(new ReasonForDefaultingModel(
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.ID)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.OUTREACH_DATE)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.FOLLOWUP_DATE)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.OUTREACH_DEFAULTING_REASON)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.ADDITIONAL_DEFAULTING_NOTES)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.BASE_ENTITY_ID)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.EVENT_DATE)),
+                            cursor.getString(cursor.getColumnIndex(GizConstants.JsonAssetsHelper.OTHER_DEFAULTING_REASON))));
+
+                } while (cursor.moveToNext());
+            }
+
+        } catch (SQLException e) {
+            Timber.e(e, "Could not add records");
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+        }
+        return reasonForDefaultingModels;
+    }
+
+    public void purgeReasonForDefaulting(String baseEntityID) {
+        try {
+            getWritableDatabase().execSQL("DELETE FROM reason_for_defaulting WHERE id ='" + baseEntityID + "'");
+        } catch (SQLException e) {
+            Timber.e(e, "Could not delete records");
+        }
+    }
+
+    public long getReasonForDefaultingCount() {
+        SQLiteDatabase db = getWritableDatabase();
+        long count = DatabaseUtils.queryNumEntries(db, TABLE_NAME);
+        db.close();
```

Comment:
Do you really need to close the database?

Reply:
Removed